logos-derive/src/parser/type_params.rs

```diff
@@ -142,8 +137,65 @@ pub fn replace_lifetimes(ty: &mut Type) {
 
             r.lifetime = Some(Lifetime::new("'s", span));
         },
-        Type::Slice(slice) => replace_lifetimes(&mut slice.elem),
-        Type::Tuple(tuple) => tuple.elems.iter_mut().for_each(replace_lifetimes),
         _ => (),
     }
 }
+
+pub fn replace_types(ty: &mut Type, f: &mut impl FnMut(&mut Type)) {
```

Comment:
Given that this function doesn't actually replace anything by itself, would `traverse_type[s]?` (ditto for `traverse_path`) be a more fitting name?